### DISCUSSION TITLE
Windows is not as fast as linux for downloading files

### DISCUSSION THREAD

**Discussion by futurecad** [slackhq/nebula#1159] [Q&A]
### What version of `nebula` are you using? (`nebula -version`)

1.9.2

### What operating system are you using?

Windows

### Describe the Bug

Windows is not as fast as linux for downloading files

When I download the same file, the download speed on the Linux client machine is 800kb, but on Windows (windows10) it is only about 400kb.

### Logs from affected hosts

```

```

### Config files from affected hosts

```

```

**Comment by johnmaguire** [comment]
Hi @futurecad -

Different platforms and workflows have different performance characteristics, and various environmental causes can affect performance as well.

Can you please provide a detailed explanation of the type of traffic you're testing with, and the exact methodology you are using to test?

**Comment by futurecad** [comment]
The test was done using Google Chrome and then downloading the resource files of the server behind Nebula. For the same file, the speed of Linux is twice that of Windows 10, and Windows 10 is twice that of Windows 7.

**Comment by futurecad** [comment]
I'm not sure where the problem is. I'm using the latest version of Nebula. In a previous question, I remember you said that Windows 10 uses Rio to improve speed.

**Comment by johnmaguire** [comment]
@futurecad Windows has less use than other platforms and it's not a daily driver for any of the Nebula maintainers, so it's difficult to say off-hand if the performance you're seeing is typical. Can you please provide the Nebula config files you're using to test with to see if anything stands out? (Windows 7, Windows 10, Linux, and the web server / file host?) For example, the `routines` directive is not supported on Windows.

> The test was done using Google Chrome and then downloading the resource files of the server behind Nebula. For the same file, the speed of Linux is twice that of Windows 10, and Windows 10 is twice that of Windows 7.

To clarify, is "the server behind Nebula" running Nebula itself, or is it exposed via `unsafe_routes`?

**Comment by futurecad** [comment]
It comes through unsafe_routes

`
pki:
  ca: |
   
  cert: |
    
  key: |
   

static_host_map:
  "11.206.1.1": ["10.139.54.12:6666"]

lighthouse:
  am_lighthouse: false
  interval: 60
  hosts:
    - "11.206.1.1"

listen:
  host: "[::]"
  port: 0

punchy:
  punch: true

relay:
  relays:
    - 11.206.1.1
  am_relay: false
  use_relays: true

tun:
  disabled: false
  dev: nebula
  drop_local_broadcast: true
  drop_multicast: true
  tx_queue: 500
  mtu: 1350

  unsafe_routes:
    - route: 10.139.54.26/32
      via: 11.206.1.1
    - route: 10.139.86.53/32
      via: 11.206.1.1


logging:
  level: info
  format: text

firewall:
  outbound_action: drop
  inbound_action: drop

  conntrack:
    tcp_timeout: 12m
    udp_timeout: 3m
    default_timeout: 10m

  outbound:
    - port: any
      proto: any
      host: any

  inbound:
    - port: any
      proto: any
      host: any
`
Above is my configuration file
I finally want to access the service of 10.139.54.26 and then download files from this service. The download speed of windwos10 is not as good as the download speed of linux.

**Comment by futurecad** [comment]
@johnmaguire In the same environment, I use wireguard on windows faster than nebula.
Does nebula integrate wireguard-nt?
